src/mpid/ch4/shm/ipc/src/ipc_init.c

```diff
@@ -38,6 +38,9 @@ int MPIDI_IPC_init_world(void)
 {
     int mpi_errno = MPI_SUCCESS;
 
+    mpi_errno = MPL_gpu_ipc_get_handle_type(&MPIDI_IPCI_global.ipc_handle_type);
+    MPIR_ERR_CHECK(mpi_errno);
+
```

Comment:
This change belongs in commit "mpl/gpu: query ipc handle type", currently it is in "ch4/ipc: implement support for fd-based ipc handles".

This is similar to `enable_ipc` in `MPL_gpu_info_t`, part of global GPU states. It's annoying to get them ad hoc. How about just keep an global `MPL_gpu_info_t MPL_gpu_info;`, filled at `gpu_init`, and then we can directly use it rather than inventing individual retrieve APIs.

Reply:
Addressed the first comment on moving to the correct commit.

I agree, I will add `ipc_handle_type` to `MPL_gpu_info_t` and make it global

Reply:
Moved `ipc_handle_type` to `MPL_gpu_info_t` with global access and removed the retrieve APIs.